### PR TITLE
[sai-ptf] set expected exception and fix errors in test

### PR DESCRIPTION
**Pull request by richardyu-ms**
# Why
## Acceptable errors
In SAI-PTF test cases, in some platforms some interface/attribute might not implemented/support.
Then we need to skip the test for unsupported errors

## syntax error
Fix syntax error when cases running into bad state for some unexpected code path

# How
Add error code and check the API state after return. Handle the errors base on defined acceptable errors.
Fix syntax error

# Test
Pipeline

### CODE REVIEW

**Comment by richardyu-ms** [comment]
@ravi861 please help review.

File: ptf/sairoute.py [hunk 1]
```diff
@@ -1084,10 +1084,12 @@ def runTest(self):
                 sai_thrift_remove_next_hop(self.client, nhop1)
             if nhop2:
                 sai_thrift_remove_next_hop(self.client, nhop2)
-
-            sai_thrift_remove_neighbor_entry(self.client, nbr_entry1)
-            sai_thrift_remove_neighbor_entry(self.client, nbr_entry2)
-            sai_thrift_remove_neighbor_entry(self.client, nbr_entry0)
+            if 'nbr_entry1' in locals():
```
Comment (ravi861) [inline ptf/sairoute.py:1087]: why do we need this specific check here?

Comment (richardyu-ms) [reply ptf/sairoute.py:1087]: Some tests failed before defining the variable ``nbr_entry1``, then there is another error that happened there.
There is the error backtrace
```
Traceback (most recent call last):
[File "/tmp/sai_qualify/ptf/sairoute.py", line 1029, in runTest](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=%2Ftmp%2Fsai_qualify%2Fptf%2Fsairoute.py&version=GBinternal&_a=contents&line=1029&lineEnd=1030&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
self.trafficTrapTest1()
[File "/tmp/sai_qualify/ptf/sairoute.py", line 882, in trafficTrapTest1](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=%2Ftmp%2Fsai_qualify%2Fptf%2Fsairoute.py&version=GBinternal&_a=contents&line=882&lineEnd=883&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
pre_stats["SAI_QUEUE_STAT_PACKETS"] + 1)
[File "/usr/lib/python3.7/unittest/case.py", line 839, in assertEqual](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=%2Fusr%2Flib%2Fpython3.7%2Funittest%2Fcase.py&version=GBinternal&_a=contents&line=839&lineEnd=840&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
assertion_func(first, second, msg=msg)
[File "/usr/lib/python3.7/unittest/case.py", line 832, in _baseAssertEqual](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=%2Fusr%2Flib%2Fpython3.7%2Funittest%2Fcase.py&version=GBinternal&_a=contents&line=832&lineEnd=833&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
raise self.failureException(msg)
AssertionError: 1 != 2
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
[File "/tmp/sai_qualify/ptf/sairoute.py", line 1088, in runTest](https://dev.azure.com/mssonic/internal/_git/5380e8f7-6e2a-4154-8dee-f3be7b096894?path=%2Ftmp%2Fsai_qualify%2Fptf%2Fsairoute.py&version=GBinternal&_a=contents&line=1088&lineEnd=1089&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
sai_thrift_remove_neighbor_entry(self.client, nbr_entry1)
```

Comment (ravi861) [reply ptf/sairoute.py:1087]: This is passing on Barefoot but we can keep the change for now.

**Review by ravi861 — APPROVED** [review]

**Pull request merged**